### DISCUSSION TITLE
Directions Q&A - How to Pave the Way For Your App Development: Submitting Code to the System Application

### DISCUSSION THREAD

**Discussion by darjoo** [microsoft/ALAppExtensions#9769] [Q&A] [answered]
Hi all,

You can find the questions that were asked at the Directions Q&A and the answers here. If you have further questions, feel free to ask them :)

------

How do we specify the range of objects we want to use? 

>We don't have a way to reserve empty object ranges. Because the objects would go into the System Application, it needs to be a non-overlapping ID between 1-10000 

What about changes to the Base Application? 

>The Base Application is not open source, so changes must go through the traditional idea channels. 

Who creates permission sets? 

>We’ll create them based on how the module is created. This can also be discussed in the pull request.  

Is it possible to use non-standard assemblies?  

>For legal reasons, non-standard assemblies will need to be reviewed on a case-by-case basis. 

Can we use this channel to submit new event publishers for modules for the Base Application? 

>Unfortunately no, but you can request event publishers on the Issues tab. 

To what extent do you judge whether a module makes sense, or whether it is more related to a partner/customer use case? 

>Generally, the System Application provides horizontal features, which means that contributions should fall within this area. Specific requests can be submitted to our discussion page on GitHub. 

How can we determine which IDs are available? 

>Through trial and error, unfortunately. We do not have a way to identify available IDs. 

As an example, I could create a pull request for a module that handles global parameters bound to a single session, would that be of interest for the System Application?  

>Specific requests can be submitted here on our discussion page on GitHub. 

How is the System Application created from small apps? 

>The System Application is an aggregation of all modules. Instead of making each module into a separate extension, it's just combined and compiled as one extension. (I hope I understood the question properly)

**Comment by JesperSchulz** [comment, marked as answer]
Closing Discussion.
